src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8004,10 +8054,9 @@ bool tryAsMemberOfSingleType(NamedTypeSymbol singleType, [NotNullWhen(true)] out
 
             (BoundExpression operand, Conversion conversion) = RemoveConversion(node, includeExplicitConversions: true);
             SnapshotWalkerThroughConversionGroup(node, operand);
-            if (targetType.SpecialType == SpecialType.System_Boolean &&
-                (operand.Type?.SpecialType == SpecialType.System_Boolean || operand.Type?.IsErrorType() == true))
+            if (TypeAllowsConditionalState(targetType.Type) && TypeAllowsConditionalState(operand.Type))
```

Comment:
FYI @RikkiGibson I integrated with the change from other PR

Reply:
This seems more permissive than the version of the check from the PR, is there any need to test when source or target type of the conversion is dynamic, for example, and we have a split state after visiting the operand?

Reply:
Added a test where a difference is observable (`PatternOnTuple_IsExpression_DynamicCast`). There would be a nullable warning before this change.